rclcpp_components/include/rclcpp_components/component_manager.hpp

```diff
@@ -60,41 +62,89 @@ class ComponentManager : public rclcpp::Node
    */
   using ComponentResource = std::pair<std::string, std::string>;
 
+  /// Default constructor
+  /**
+   * Initializes the component manager. It creates the services: load node, unload node
+   * and list nodes.
+   *
+   * \param executor the executor which will spin the node.
+   * \param node_name the name of the node that the data originates from.
+   * \param node_options additional options to control creation of the node.
+   */
   RCLCPP_COMPONENTS_PUBLIC
   ComponentManager(
     std::weak_ptr<rclcpp::Executor> executor,
     std::string node_name = "ComponentManager",
     const rclcpp::NodeOptions & node_options = rclcpp::NodeOptions());
 
+  /// Default destructor
   RCLCPP_COMPONENTS_PUBLIC
   virtual ~ComponentManager();
 
-  /// Return a list of valid loadable components in a given package.
+  /// Get the component resource
+  /*
+   * \param package_name name of the package
+   * \param resource_index name of the executable
+   * \throws ComponentManagerException if the resource was not found or a invalid resource entry
+   * \return a list of component resources
+   */
   RCLCPP_COMPONENTS_PUBLIC
   virtual std::vector<ComponentResource>
   get_component_resources(
     const std::string & package_name,
     const std::string & resource_index = "rclcpp_components") const;
 
+  /// Create the component factory
+  /*
+   * \param resource list of component resources
+   */
   RCLCPP_COMPONENTS_PUBLIC
   virtual std::shared_ptr<rclcpp_components::NodeFactory>
   create_component_factory(const ComponentResource & resource);
 
 protected:
+  // Service to load a new node in the component
```

Comment:
This is the service callback, right? Say that instead. Also, when might a deriving class want to override these functions?

These questions apply also to the next two functions.

Reply:
If you derive from this class and change these methods we are going to change the behaviour of this abstraction. do you think we should explicitly indicate it?

Reply:
I was just surprised to see protected virtual callbacks, and I wasn't entirely sure when that would be useful to override, so I thought some documentation might help. But seeing as all the public functions are virtual as well, I guess this class was just designed with that sort of flexibility in mind.